include/seqan3/range/view/detail.hpp

```diff
@@ -13,301 +13,296 @@
-//  view_base
+//  forwards
 // ============================================================================
 
-/*!\brief An empty base class that our views inherit from so they are treated as views by the STL/range-v3.
- * \ingroup view
- */
-struct view_base : public ranges::view_base
-{};
+template <typename left_adaptor_t, typename right_adaptor_t>
+class combined_adaptor;
 
 // ============================================================================
-//  pipable_adaptor_base
+//  adaptor_base
 // ============================================================================
 
-/*!\brief   A CRTP base class for implementing the pipable behaviour of views.
+/*!\brief CRTP-base to simplify the definition of range adaptor closure objects and similar types.
+ * \tparam derived_type CRTP specialisation parameter.
+ * \tparam stored_args_ts Some closure objects store values, these are their types.
  * \ingroup view
- * \tparam  derived_type The CRTP specialisation, must provide a `impl()`-member.
  *
  * \details
  *
- * ### Background
+ * See \ref howto_write_a_view for details on what range adaptor closure objects are.
+ *
+ * This class requires from the `derived_type`:
+ *   * an `impl()` function that takes a range and possibly further arguments.
+ *
+ * This base class provides:
+ *   * Storage of args, including perfect forwarding and moving-out.
+ *   * `operator()` that takes a range and passes the range and stored args to the derived_type.
+ *   * `operator|` that also takes a range and passes the range and stored args to the derived_type.
+ *   * `operator|` that takes another adaptor and returns seqan3::detail::combined_adaptor
+ *
+ * # Examples
  *
- * A full view implementation consists of three entities:
+ * ## "Direct derivation"
  *
- *   1. the actual view, e.g. `view_foo`;
- *   2. an adaptor that returns instances of 1. and enables usage in pipes, e.g. `foo_fn`
- *   3. an instance of the adaptor, e.g. `view::foo` that is usable by consumers of your view
+ * From include/seqan3/range/view/persist.hpp:
  *
- * For more details, see the HowTo on writing views in the SeqAn3 wiki: https://github.com/seqan/seqan3/wiki
+ * \snippet include/seqan3/range/view/persist.hpp adaptor_def
  *
- * ### Details
+ * This adaptor directly derives from adaptor_base so that it can decide between
+ * actually calling the respective view's constructor and delegating to a different view
+ * (in this case `std::view::all`) based on the type of input.
  *
- * This CRTP base class can be used to simplify the implementation of the adaptor type. For most use cases it is
- * sufficient to just use the specialisation seqan3::detail::generic_pipable_view_adaptor, but other specialisations
- * like seqan3::detail::deep_view_adaptor_wrapper also make use of this base template.
+ * ## Derived templates
+ *
+ * See `seqan3::detail::combined_adaptor`, `seqan3::detail::adaptor_for_view_without_args` and
```

Comment:
Sometimes we do this, sometimes we don't. But I think we should do it everywhere. It's more consistent, because "it is C++" and not "description text". Also it highlights broken links.